### PR TITLE
Eventhandler additions

### PR DESCRIPTION
**Pull request by MaGaMeGa**

### CODE REVIEW

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java [hunk 1]
```diff
@@ -407,6 +409,111 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
...
+	 */
+	public void subscribeToEventHandler(final SubscriptionRequestDTO request) {
+
+		final CoreServiceUri uri = getCoreServiceUri(CoreSystemService.EVENT_SUBSCRIBE_SERVICE);
+		if (uri == null) {
+			logger.debug("Subscription couldn't be proceeded due to the following reason: " +  CoreSystemService.EVENT_SUBSCRIBE_SERVICE.name() + " not known by Arrowhead Context");
+			return;
+		}
+		
+		httpService.sendRequest( Utilities.createURI( getUriScheme(), uri.getAddress(), uri.getPort(), uri.getPath() ), HttpMethod.POST, Void.class, request );
```
Comment (borditamas) [inline src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: Remove unnecessary spaces after and before the brackets

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java [hunk 2]
```diff
@@ -407,6 +409,111 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
...
+	 * @param subscriberName String which represents the required subscriberName parameter of the http(s) request
+	 * @param subscriberAddress String which represents the required subscriberAddress parameter of the http(s) request
+	 * @param subscriberPort int which represents the required subscriberPort parameter of the http(s) request
+	 * @throws AuthException when you are not authorized by Event Handler Core System
+	 * @throws BadPayloadException when the payload couldn't be validated by Event Handler Core System 
+	 * @throws InvalidParameterException when the payload content couldn't be validated by Event Handler Core System
+	 * @throws ArrowheadException when internal server error happened at Event Handler Core System
+	 * @throws UnavailableServerException when Event Handler Core System is not available
+	 */
+	public void unsubscribeFromEventHandler(
```
Comment (borditamas) [inline src/main/java/eu/arrowhead/client/library/ArrowheadService.java:448]: Remove unnecessary spaces after and before the brackets

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java [hunk 3]
```diff
@@ -407,6 +409,111 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
...
+	 * 
+	 * @returns Arrowhead Client-System ServerCN
+	*/
+	public String getServerCN() {
+
+		try {
+			
+			return (String) arrowheadContext.get( CommonConstants.SERVER_COMMON_NAME );
+
+		} catch (Exception ex) {
```
Comment (borditamas) [inline src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: final

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java [hunk 4]
```diff
@@ -407,6 +409,111 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
...
+	 */
+	public void publishToEventHandler(final EventPublishRequestDTO request) {
+
+		final CoreServiceUri uri = getCoreServiceUri(CoreSystemService.EVENT_PUBLISH_SERVICE);
+		if (uri == null) {
+			logger.debug("Publishing couldn't be proceeded due to the following reason: " +  CoreSystemService.EVENT_PUBLISH_SERVICE.name() + " not known by Arrowhead Context");
+			return;
+		}
+		
+		httpService.sendRequest( Utilities.createURI( getUriScheme(), uri.getAddress(), uri.getPort(), uri.getPath() ), HttpMethod.POST, Void.class, request );
```
Comment (borditamas) [inline src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: Remove unnecessary spaces after and before the brackets

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java [hunk 5]
```diff
@@ -407,6 +409,111 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
...
+	 */
+	public void unsubscribeFromEventHandler(
+			final String eventType,
+			final String subscriberName,
+			final String subscriberAddress,
+			final int subscriberPort ) {
+
+		final CoreServiceUri uri = getCoreServiceUri(CoreSystemService.EVENT_UNSUBSCRIBE_SERVICE);
+		if (uri == null) {
+			logger.debug("Unsubscription couldn't be proceeded due to the following reason: " +  CoreSystemService.EVENT_SUBSCRIBE_SERVICE.name() + " not known by Arrowhead Context");
```
Comment (borditamas) [inline src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: CoreSystemSerice should be UNsubscribe

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java [hunk 6]
```diff
@@ -407,6 +409,111 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
...
+		httpService.sendRequest( Utilities.createURI( getUriScheme(), uri.getAddress(), uri.getPort(), uri.getPath() ), HttpMethod.POST, Void.class, request );
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Get the serverCN from arrowheadContext
+	 * 
+	 * @returns Arrowhead Client-System ServerCN
+	*/
+	public String getServerCN() {
```
Comment (borditamas) [inline src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: Remove unnecessary spaces after and before the brackets

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java [hunk 7]
```diff
@@ -407,6 +409,111 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
...
+
+		try {
```
Comment (borditamas) [inline src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: I think here we shouldn't catch the exceptions,  but let them happen,. When someone needs the CN and it is not in the context, then it is a programming mistake. 
This implementation should be only 
"return (String) arrowheadContext.get( CommonConstants.SERVER_COMMON_NAME );"

**Review by borditamas — APPROVED** [review]
See my inline comments. After correcting them you can merge it from my side.

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java (hunk 1 above)
Comment (MaGaMeGa) [reply src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: fixed

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java (hunk 2 above)
Comment (MaGaMeGa) [reply src/main/java/eu/arrowhead/client/library/ArrowheadService.java:448]: fixed

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java (hunk 4 above)
Comment (MaGaMeGa) [reply src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: fixed

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java (hunk 5 above)
Comment (MaGaMeGa) [reply src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: fixed

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java (hunk 7 above)
Comment (MaGaMeGa) [reply src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: fixed

File: src/main/java/eu/arrowhead/client/library/ArrowheadService.java (hunk 6 above)
Comment (MaGaMeGa) [reply src/main/java/eu/arrowhead/client/library/ArrowheadService.java]: fixed

**Pull request merged**